docs/docs/concepts/multi_agent.md

```diff
@@ -26,54 +26,173 @@ There are several ways to connect agents in a multi-agent system:
 - **Hierarchical**: you can define a multi-agent system with [a supervisor of supervisors](https://langchain-ai.github.io/langgraph/tutorials/multi_agent/hierarchical_agent_teams/). This is a generalization of the supervisor architecture and allows for more complex control flows.
 - **Custom multi-agent workflow**: each agent communicates with only a subset of agents. Parts of the flow are deterministic, and only some agents can decide which other agents to call next.
 
+### Handoffs
+
+In multi-agent architectures agents can be represented as graph nodes. Each agent node executes its step(s) and decides what to do next — finish execution or route to another agent (including routing to itself, e.g. running in a loop). A common pattern in multi-agent interactions is **handoffs**: one agent handing off control to another agent. Handoffs allow you to specify:
```

Comment:
Let's generalize this -- our handofss don't need to go to an agent, they can go to any node

Reply:
discussed offline to keep as is